pkg/capture/provider/network_capture_unix.go

```diff
@@ -361,16 +364,44 @@ func (ncp *NetworkCaptureProvider) Cleanup() error {
 	return nil
 }
 
-func obtainIptablesMode() string {
+type iptablesMode string
+
+const (
+	legacyIptablesMode iptablesMode = "legacy"
+	nftIptablesMode    iptablesMode = "nft"
+)
+
+func obtainIptablesMode(l *log.ZapLogger) iptablesMode {
```

Comment:
I think you should return unhandled errors along with mode.

Reply:
The ideas in my mind is the failed iptables rules does not break the whole capture.